### PR TITLE
Add role property binding to alert

### PR DESCRIPTION
**Pull request by eduardh-landg**
# Description

This PR adds functionality to set / override the role property on an alert.  

# Checklist:

- [x] The commit messages follow the [convention for this project](./blob/master/docs/CONTRIBUTING.md#conventional-commits)
- [x] I have provided an adequate amount of test coverage
- [ ] I have added the functionality to the [test app](./blob/master/docs/CONTRIBUTING.md#build-test-application)
- [x] I have provided a story in storybook to document the changes
- [x] I have added the documentation
- [ ] I have added any new public feature modules to public-api.ts

### CODE REVIEW

**Review by markblandford — APPROVED** [review]
I think this is great. It provides us with the ability to set the `role` attribute ourselves, without which it is simply not possible (unless info or generic type of alert).

File: projects/canopy/src/lib/alert/alert.component.ts [hunk 1]
```diff
@@ -30,20 +30,30 @@ export class LgAlertComponent {
...
+    this.variant = 'generic';
   }
 
-  constructor(private renderer: Renderer2, private hostElement: ElementRef) {
-    this.variant = 'generic';
+  @Input() set role(role: string) {
+    this._role = role;
+  }
+  private setRole() {
+    if (!this._role) {
```
Comment (pauleustice) [inline projects/canopy/src/lib/alert/alert.component.ts:53]: Could we combine the nested `if` conditions? I can't see any value to separating them in this case.

**Review by pauleustice — CHANGES_REQUESTED** [review]
Looks good, thanks for the contribution! Could you please add the new input to `libs/alert/docs/alert.stories.ts` too, though?

That will allow others to test the behaviour in Storybook

File: projects/canopy/src/lib/alert/alert.component.ts (hunk 1 above)
Comment (eduardh-landg) [reply projects/canopy/src/lib/alert/alert.component.ts:53]: Sure thing!

Comment (markblandford) [reply projects/canopy/src/lib/alert/alert.component.ts:53]: I'm not sure I completely agree. I see this `if` as being a sort of 'guard'. I think it's possibly easier to read too, I'm not keen on `if` statements containing multiple logical operators (especially when dealing with Not).

Comment (Xinchro) [reply projects/canopy/src/lib/alert/alert.component.ts:53]: if we're going to keep it separate to escape early, would it be better to make it explicit and have something along the lines of...
```
private setRole() { 
  if (!this._role) return

  if...```

Comment (eduardh-landg) [reply projects/canopy/src/lib/alert/alert.component.ts:53]: https://github.com/Legal-and-General/canopy/blob/c2d78bfb37bbc4ccfa0a495b341a253f2deee2e5/projects/canopy/src/lib/accordion/accordion-item/accordion-item.component.ts#L117

All roads lead to Rome. I merged the if clauses as per the initial request.

**Review by markblandford — APPROVED** [review]

File: projects/canopy/src/lib/alert/alert.component.ts (hunk 1 above)
Comment (eduardh-landg) [reply projects/canopy/src/lib/alert/alert.component.ts]: @pauleustice I also added role input in storybook.

**Review by pauleustice — APPROVED** [review]
Good stuff - tested in Storybook deployment and all looks good!

**Review by markblandford — APPROVED** [review]

**Review by owensgit — APPROVED** [review]

**Review by Xinchro — APPROVED** [review]

**Pull request merged**